infra/modules/azure_postgres_server/variables.tf

```diff
@@ -0,0 +1,271 @@
+variable "default_metric_alerts" {
+
+  description = <<EOD
+  Map of name = criteria objects
+  EOD
+
+  type = map(object({
+    # criteria.*.aggregation to be one of [Average Count Minimum Maximum Total]
+    aggregation = string
+    metric_name = string
+    # "Insights.Container/pods" "Insights.Container/nodes"
+    metric_namespace = string
+    # criteria.0.operator to be one of [Equals NotEquals GreaterThan GreaterThanOrEqual LessThan LessThanOrEqual]
+    operator  = string
+    threshold = number
+    # Possible values are PT1M, PT5M, PT15M, PT30M and PT1H
+    frequency = string
+    # Possible values are PT1M, PT5M, PT15M, PT30M, PT1H, PT6H, PT12H and P1D.
+    window_size = string
+    # severity: The severity of this Metric Alert. Possible values are 0, 1, 2, 3 and 4. Defaults to 3.
+    severity = number
+  }))
+
+  default = {
+    cpu_percent = {
+      frequency        = "PT5M"
+      window_size      = "PT30M"
+      metric_namespace = "Microsoft.DBforPostgreSQL/flexibleServers"
+      aggregation      = "Average"
+      metric_name      = "cpu_percent"
+      operator         = "GreaterThan"
+      threshold        = 80
+      severity         = 2
+    },
+    memory_percent = {
+      frequency        = "PT5M"
+      window_size      = "PT30M"
+      metric_namespace = "Microsoft.DBforPostgreSQL/flexibleServers"
+      aggregation      = "Average"
+      metric_name      = "memory_percent"
+      operator         = "GreaterThan"
+      threshold        = 80
+      severity         = 2
+    },
+    storage_percent = {
+      frequency        = "PT5M"
+      window_size      = "PT30M"
+      metric_namespace = "Microsoft.DBforPostgreSQL/flexibleServers"
+      aggregation      = "Average"
+      metric_name      = "storage_percent"
+      operator         = "GreaterThan"
+      threshold        = 80
+      severity         = 2
+    },
+    active_connections = {
+      frequency        = "PT5M"
+      window_size      = "PT30M"
+      metric_namespace = "Microsoft.DBforPostgreSQL/flexibleServers"
+      aggregation      = "Average"
+      metric_name      = "active_connections"
+      operator         = "GreaterThan"
+      threshold        = 80
+      severity         = 2
+    },
+    connections_failed = {
+      frequency        = "PT5M"
+      window_size      = "PT30M"
+      metric_namespace = "Microsoft.DBforPostgreSQL/flexibleServers"
+      aggregation      = "Total"
+      metric_name      = "connections_failed"
+      operator         = "GreaterThan"
+      threshold        = 80
+      severity         = 2
+    }
+  }
+}
+
+variable "alerts_enabled" {
+  type        = bool
+  default     = false
+  description = "Define if alerts should be enabled."
+}
+
+variable "alert_action" {
+  description = "The ID of the Action Group and optional map of custom string properties to include with the post webhook operation."
+  type = set(object(
+    {
+      action_group_id    = string
+      webhook_properties = map(string)
```

Comment:
I suggest to remove `webhook_properties` as we did for [Function Apps](https://github.com/pagopa/dx/blob/main/infra/modules/azure_function_app/alerts.tf) as it is never used